lib/monitoring/cloudwatch.js

```diff
@@ -8,22 +8,25 @@ module.exports.ADMITHUB_DEV_SRN = 'arn:aws:sns:us-east-1:962178857523:ai-matcher
 
 module.exports.getCloudWatchClient = ({awsConfigFile = AWS_CONFIG_FILENAME}) => {
   let awsConfig;
-  if (fs.existsSync(__dirname + '/' + awsConfigFile)) {
-    awsConfig = require(AWS_CONFIG_FILENAME);
-  } else if (process.env.AWS_ACCESS_KEY_ID && process.env.AWS_SECRET_ACCESS_KEY) {
-    awsConfig = {
-  	  'accessKeyId': process.env.AWS_ACCESS_KEY_ID,
-  	  'secretAccessKey': process.env.AWS_SECRET_ACCESS_KEY
-  	};
-  } else {
-	  throw new Error(`No AWS config in file ${AWS_CONFIG_FILENAME} or in environment variables`);
-  }
+  const environment = process.env.NODE_ENV
+  if (environment && environmet !== 'testing') {
+    if (fs.existsSync(__dirname + '/' + awsConfigFile)) {
+      awsConfig = require(AWS_CONFIG_FILENAME);
+    } else if (process.env.AWS_ACCESS_KEY_ID && process.env.AWS_SECRET_ACCESS_KEY) {
+      awsConfig = {
+    	  'accessKeyId': process.env.AWS_ACCESS_KEY_ID,
+    	  'secretAccessKey': process.env.AWS_SECRET_ACCESS_KEY
+    	};
+    } else {
+  	  throw new Error(`No AWS config in file ${AWS_CONFIG_FILENAME} or in environment variables`);
+    }
 
-  AWS.config.update({
-    accessKeyId: awsConfig.accessKeyId,
-	  secretAccessKey: awsConfig.secretAccessKey,
-	  region: AWS_REGION
-	});
+    AWS.config.update({
+      accessKeyId: awsConfig.accessKeyId,
+  	  secretAccessKey: awsConfig.secretAccessKey,
+  	  region: AWS_REGION
+  	});
 
-  return new AWS.CloudWatch({});
+    return new AWS.CloudWatch({});
+  }
```

Comment:
It looks like the function returns nothing when there is no non-test environment. Can we throw an exception instead?

Reply:
For some reason, when we test now in NeOliTh, `process.env.NODE_ENV` is undefined. The way environment variables are set in NeOliTh has been recently changed around by Noah, and was kind of complicated to begin with (we're using what we inherited from Oli), so I was hoping to avoid opening that can of worms, but maybe it's worth figuring it out -- what do you think?

Reply:
During test can you use the MockPublisher?

Reply:
It's just `isMock=true`

Reply:
I haven't tried; I'm actually not sure how I would do that, I'm embarrassed to say. I was planning on live testing it. `process.env.NODE_ENV` is definitely "staging" on the neolith staging server. I'd be interested to know if you have ideas on unit tests.

Reply:
How about we use the mock publisher for everything that is *not* prod?

Reply:
I'm actually using this code up in NeOliTh:
```
  makeHeartbeat: async () => {
    const environment = conf.environment
    const isMock = environment !== 'production'
    const heartbeatParams = {
      'appName': 'NeOliTh',
      'appStage': environment,
      'hearbeatName': 'NeOliTh Heartbeat',
      'period': 5.0,
      isMock
    }
    const heartbeat = await new HeartbeatMonitor(heartbeatParams)
    return heartbeat
```
On reflection, maybe that means this cloudwatch code doesn't get called unless we're in a production environment, in which case this if block wrapper is unnecessary. Is that right? I'll check.

Reply:
I think that's right, in which case you should just close this pull request. Thanks comrade.

Reply:
Yes, that seems correct to me.